### PR TITLE
Support openAPI 3.0

### PR DESCRIPTION
**Pull request by joeb1415**

### CODE REVIEW

**Comment by joeb1415** [comment]
@barakalon Got a start on this. Bunch more to do but I'll be out the rest of the week so figured I'd see if you had any thoughts to share on it so far.  I like the SwaggerObject classes. A little silly in places when I need to use getattr/setattr, but I'm finding using it helpful to understand the spec as I go. 

This has been fun so far I hope to continue on it next week when I'm back!

File: flask_rebar/swagger_generation/swagger_generator.py [hunk 1]
```diff
@@ -668,18 +655,303 @@ def _get_definitions(self, paths):
...
+
+        required = obj.get('required', [])
+
+        for name, prop in sorted(obj['properties'].items(), key=lambda i: i[0]):
+            schema = oa.Schema()
+            for k, v in prop.items():
+                python_key = _get_python_key_from_swagger_word(k)
+                setattr(schema, python_key, v)
+
+            parameter = oa.Parameter(
```
Comment (barakalon) [inline flask_rebar/swagger_generation/swagger_generator.py]: I don't think you can instantiate marshmallow schemas like this - `Schema` may not be the object you're looking for.

You might want to opt for regular ol' classes here. Or maybe get fancy with a new type.

**Comment by barakalon** [comment]
Sweet!

Yeah, as mentioned before, I think learning the intricacies of OpenAPI 3.0 is the hardest part here, and it looks like you're doing that 👍

**Comment by RookieRick** [comment]
resolves #18

**Comment by Sytten** [comment]
Any progress in view for this?

**Comment by joeb1415** [comment]
Haven't had much chance to get back into it. Did just yesterday go over these changes so far with @RookieRick who might be picking this up.

**Comment by RookieRick** [comment]
I did get green light to put some effort towards this given potential
increasing demand internally as well. I don't have any timeline yet but
will definitely be looking to show this some love in the very near future.
😁

**Comment by Sytten** [comment]
I guess we can close that now @RookieRick :)

**Comment by RookieRick** [comment]
Closing this one; @joeb1415 if there are things in this that would augment our existing implementation let's salvage in new PRs/issues

**Pull request closed without merging**